octodns/provider/yaml.py

```diff
@@ -111,7 +111,8 @@ class YamlProvider(BaseProvider):
                     'URLFWD'))
 
     def __init__(self, id, directory, default_ttl=3600, enforce_order=True,
-                 populate_should_replace=False, *args, **kwargs):
+                 populate_should_replace=False, supports_root_ns=True,
```

Comment:
What is the purpose of offering the `supports_root_ns` toggle?

Reply:
It's mostly a convenience for testing so that 75% of the tests in this repo and the providers don't have to change if it's not relevant to what's at test. 

It is also an option for people running octoDNS if they'd like to ignore the root NS records that are in their config, thought I'd be a bit surprised/disappointed if anyone uses it.